### ISSUE TITLE
va-pagination: Page link aria-labels fail SC 2.5.3 Label in Name

### ISSUE THREAD

**Issue by coforma-jamie** [department-of-veterans-affairs/vets-design-system-documentation#4973]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

Using MacOS Voice Control, I tried to access the third page of pagination by saying `Click 3`, but it didn't work.

This is because each numbered page link in the pagination component has an aria-label starting with the word "Page," such as "Page 3":
`<a href="javascript:void(0)" class="usa-pagination__button" aria-label="page 3">3</a>`

This is a violation of [SC 2.5.3 Label in Name](https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html), which requires that the accessible name and visible label start with the same characters, in this case, the page number.

Since "3 page" doesn't make much sense in English, I recommend:
- changing the first page aria-label to "1 of [number of pages] pages, first page" (instead of "page 1, first page")
- changing the last page aria-label to "[number] of [number of pages] pages, last page" (instead of "page [number], last page")
- changing all other aria-labels from "Page [number]" to "[number] of [number of pages] pages" (i.e., "2 of 10 pages")

## What I expected to happen

To be able to select a page in pagination via MacOS Voice Control, giving the command "Click [page number]"

## Reproducing

- CSS Library version:
- Device: Macbook
- Browser: Chrome

**URL(s) where this bug occurs**
All instances of va-pagination - you can test it here https://design.va.gov/storybook/iframe.html?globals=&args=&id=uswds-va-pagination--default&viewMode=story

**Steps to reproduce:**

1. Turn on Voice Control
2. If you're seeing the number flags, say "Hide numbers"
3. "Say 'Click 3'" to attempt to select page 3 in the pagination component

## Urgency

How urgent is this request? Please select the appropriate option below and/or provide details

- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [x] This bug is not blocking any work
- [ ] Other

### Details
This is also an issue in the USWDS
